feat(startproject): add support to indicate Godot project directory.

SocratesDz opened the pull request:
Add `--godot-dir` option on `startproject` command to specify the Godot project directory where the `.gdextension` file will be saved. It's useful in case you want to separate your Godot project from the Rust project. It also specifies the libraries paths as well so this doesn't need to be tweaked by the user.

Usage:

```bash
# pwd -> ~/my-new-project/rust
gdext-cli startproject --godot-dir ../godot example-game example-game-name

SocratesDz committed:
feat(startproject): add support to indicate Godot project directory.

Add --godot-dir option on startproject command to specify the Godot
project directory where the .gdextension file will be saved. It's useful
in case you want to separate your godot project from the rust project.
It also specifies the libraries paths as well so this doesn't need to be
tweaked by the user.

SocratesDz committed:
fix(startpage): fix a relative path issue on the .gdextension file content

Paths written to the .gdextension file had to be canonicalized to get
ride of unwanted relative symbols.

FrankCasanova commented:
Thats nice 🙂 
Is something will be useful for future features using RataUI

Ill test how it works, after that ill merge

Thanks a lot!✨

SocratesDz commented:
No problem! This tool is pretty handy to bootstrap godot-rust projects. However since I wrote it, I've been thinking that I'm probably missing the edge case of paths separators being different on Windows.

Also I need to update the README and the `--help` output.

FrankCasanova commented:
> No problem! This tool is pretty handy to bootstrap godot-rust projects. However since I wrote it, I've been thinking that I'm probably missing the edge case of paths separators being different on Windows.
> 
> Also I need to update the README and the `--help` output.

ok, let me know when everithing its ok, there are no hurry.

my goal is to turn this app for something more user-friendly using RataUI.
where the app ask to the user, and the user only need to type answer or select several options to set up the project or create a new script.
ur feature will be handy cos i think the first question the app should make is 
"Where's the Godot project directory" (default ../godot) 
or something like that, and then pass that info to ur feat :)

unless gdex project implement something to create scripts and setup the project directly from godot editor, that would be nice but so far from my knoledge to implement it haha

SocratesDz committed:
fix(startpage): sanitize OS-specific path so it's compatible with systems like Windows

SocratesDz committed:
docs(startproject): add documentation on the new --godot-dir option

SocratesDz commented:
All done. It can be reviewed now :+1:

FrankCasanova commented:
did i do something wrong?

it raise this error:
Error: paths contain different absolute prefixes

<img width="773" height="576" alt="image" src="https://github.com/user-attachments/assets/16bb213b-4e15-4154-ad6a-cf44084c9e20" />

SocratesDz commented:
> did i do something wrong?

You didn't. I'll boot up a Windows VM to see what's happening in that case.

SocratesDz commented:
@FrankCasanova Can you try also using `..\godot` instead of `../godot`? :thinking:

FrankCasanova commented:
> @FrankCasanova Can you try also using `..\godot` instead of `../godot`? 🤔

let me check

FrankCasanova commented:
@SocratesDz 
<img width="1231" height="912" alt="image" src="https://github.com/user-attachments/assets/0949c5ca-43ed-4bf2-9301-740baab48f27" />

FrankCasanova committed:
Commit message: Improve cross-platform path handling for Godot projects

- Canonicalize Godot project directory path for consistent resolution
- Normalize path separators in GDExtension configuration
- Use explicit path handling to ensure reliable relative path calculation
- Fix Windows path compatibility by replacing backslashes with slashes

FrankCasanova commented:
@SocratesDz 
this changes works perfect for me, let me know if still working well in ur machine 

:)

SocratesDz committed:
feat(startproject): use main separator constant for sanitizing paths

SocratesDz commented on src/startproject_gen.rs:
```diff
@@ -1,70 +1,123 @@
+        .canonicalize()?
+        .relative_to(&godot_project_dir)?
+        .into_string()
+        .replace(std::path::MAIN_SEPARATOR_STR, "/"); // Normalize path separators
```
Looks much better than my approach! In this case, let's use the MAIN_SEPARATOR_STR constant to future-proof things.

FrankCasanova commented:
thanks @SocratesDz 
have a nice day and good luck on everything!
